controllers/operator_controller.go

```diff
@@ -47,10 +49,33 @@ type OperatorReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.1/pkg/reconcile
 func (r *OperatorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	log := log.FromContext(ctx).WithName("reconcile")
+
+	var operator operatorsv1alpha1.Operator
+	if err := r.Get(ctx, req.NamespacedName, &operator); err != nil {
+		log.Error(err, "unable to fetch Operator")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	readyCondition := metav1.Condition{
+		Type:               operatorsv1alpha1.TypeReady,
+		Status:             metav1.ConditionFalse,
+		Reason:             "startUp",
+		Message:            "Operator startup",
+		ObservedGeneration: operator.GetGeneration(),
+	}
+	apimeta.SetStatusCondition(&operator.Status.Conditions, readyCondition)
+	log.Info("set not ready Condition")
 
 	// TODO(user): your logic here
 
+	// This is something that needs to happen if any Status is updated,
+	// even if an error occurs
+	if err := r.Status().Update(ctx, &operator); err != nil {
+		log.Error(err, "unable to update Operator status")
```

Comment:
No need to log this error. By default, the controller will log errors returned from `Reconcile`

Reply:
In this case, `err` isn't returned. It can be, but I didn't want to override any other error condition that may be present. (The Reconcile function isn't really set up this way right now.)